### PR TITLE
Basic structure for the frontpage module

### PR DESCRIPTION
**Pull request by jszanto**
Just the basic structure and a bit of functionality. Might be handy to merge this already to prevent future merge conflicts.

### CODE REVIEW

File: module/Application/src/Application/View/Helper/Acl.php [hunk 1]
```diff
@@ -0,0 +1,81 @@
+<?php
+
+namespace Application\View\Helper;
+
+use Zend\View\Helper\AbstractHelper;
+
+
+class Acl extends AbstractHelper
```
Comment (kokx) [inline module/Application/src/Application/View/Helper/Acl.php]: This was in a different PR as well!

File: module/Decision/src/Decision/Service/Member.php [hunk 2]
```diff
@@ -87,7 +87,8 @@ public function findMemberByLidNr($lidnr)
      */
     public function getBirthdayMembers($days = 0)
     {
-        if (!$this->isAllowed('birthdays')) {
+        // Allow everyone to view birthdays of today.
+        if ($days > 0 && !$this->isAllowed('birthdays')) {
```
Comment (kokx) [inline module/Decision/src/Decision/Service/Member.php]: You should actually add a privilege for showing today's birthdays, and check for that privilige for the case of `$days == 0`

File: module/Frontpage/config/module.config.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+                        'controller'    => 'Frontpage',
+                        'action'        => 'home',
+                    ),
+                ),
+                'may_terminate' => true,
+                'child_routes' => array(
+                    'page' => array(
+                        'type'    => 'Segment',
+                        'options' => array(
+                            'route'    => '/[:category[/:name]]',
```
Comment (kokx) [inline module/Frontpage/config/module.config.php]: Since this is a child route of `/`, it shouldn't start with a `/`. Also, the category wouldn't be optional for this route. And defaults should be created for the optional part. And a controller and action still have to be determined.

Also, since it kinda is a catch-all route, you should set a priority.

File: module/Frontpage/src/Frontpage/Service/Frontpage.php [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        $birthdayInfo = $this->getBirthdayInfo();
+        return array(
+            'birthdays' => $birthdayInfo['birthdays'],
+            'birthdayPhoto' => $birthdayInfo['photo']
+        );
+    }
+
+    /**
+     * Retrieves all birthdays happening today, which should be shown on the home page.
+     * Includes the age and a random photo of someone whom has a birthday.
```
Comment (kokx) [inline module/Frontpage/src/Frontpage/Service/Frontpage.php]: It shouldn't be a random photo. It should be the photo of the most active member (actually, the member with the most photos).

File: module/Frontpage/view/frontpage/frontpage/home.phtml [hunk 5]
```diff
@@ -0,0 +1,39 @@
+<div class="jumbotron">
+    <a href="<?= $this->url('home') ?>">
+        <img src="<?= $this->basePath() ?>/img/logo.png"/>
+    </a>
+    <?php /*
+    <h1><?= sprintf($this->translate('Welcome to %sGEWIS%s'), '<span class="gewis-red">', '</span>') ?></h1>
+    <p><?= sprintf($this->translate('Welcome to the website of Study association %sGEWIS%s. This is just a bootstrap thingy for now, until the design is implemented.'), '<span class="gewis-red">', '</span>') ?></p>
+    <p><a class="btn btn-success btn-lg" href="https://github.com/GEWIS/gewisweb" target="_blank"><?php echo $this->translate('Fork the GEWIS website on GitHub') ?> &raquo;</a></p>
+ */ ?>
+</div>
```
Comment (kokx) [inline module/Frontpage/view/frontpage/frontpage/home.phtml]: This bit has changed a lot in the design. See the PR on the design for this one!

File: module/Photo/Module.php [hunk 6]
```diff
@@ -95,6 +95,26 @@ public function getServiceConfig()
...
+                    $acl->allow('guest', 'album', 'view');
+
+                    // Users are allowed to view, remove and add tags
+                    $acl->allow('user', 'tag', array('view', 'add', 'remove'));
+
+                    // Users are allowed to download photos
+                    $acl->allow('user', 'photo', array('download', 'view_metadata'));
+
+                    return $acl;
+                },
```
Comment (kokx) [inline module/Photo/Module.php]: These changes really belong in a different PR. Definitely not in one just for the frontpage.

File: module/Frontpage/config/module.config.php (hunk 3 above)
Comment (jszanto) [inline module/Frontpage/config/module.config.php]: This will be changed in the future once the page part is implemented. (Actually it is changed already on another branch.)

Comment (kokx) [inline module/Frontpage/config/module.config.php]: In that case, it is bad to have the catch-all at all right now. Because it may cause weird bugs with route matching.

File: module/Photo/Module.php (hunk 6 above)
Comment (jszanto) [inline module/Photo/Module.php]: This one is based on the same branch as PR #162 which is why all the changes of PR #162 also get included here. This one probably should wait until PR #162 is resolved.

**Comment by jszanto** [comment]
oops
